app/src/app/index.jsx

```diff
@@ -0,0 +1,137 @@
+const ProfilePageAuth = userIsAuthenticated(ProfilePage)
+const DashboardListAuth = userIsAuthenticated(DashboardList)
+const DashboardEditorAuth = userIsAuthenticated(DashboardEditor)
+const StreamShowViewAuth = userIsAuthenticated(StreamShowView)
+const StreamCreateViewAuth = userIsAuthenticated(StreamCreateView)
+const StreamListViewAuth = userIsAuthenticated(StreamListView)
+const StreamLivePreviewAuth = userIsAuthenticated(StreamLivePreview)
+const TransactionListAuth = userIsAuthenticated(TransactionList)
+const PurchasesPageAuth = userIsAuthenticated(PurchasesPage)
+const ProductsPageAuth = userIsAuthenticated(ProductsPage)
+const CanvasEditAuth = userIsAuthenticated(CanvasEdit)
+
+// Other components
+const ProductPurchasePage = (props) => <ProductPage overlayPurchaseDialog {...props} />
+const ProductPublishPage = (props) => <ProductPage overlayPublishDialog {...props} />
+
+// Wrap each Route to an ErrorBoundary
+const Route = withErrorBoundary(ErrorPageView)(RouterRoute)
+
+const { userpages, docs, marketplace, internalLogin } = links
+
+const DocsRouter = () => ([
+    <Route exact path={docs.main} component={GettingStartedPage} key="GettingStartedPage" />,
+    <Route exact path={docs.introduction} component={IntroductionPage} key="IntroductionPage" />,
+    <Route exact path={docs.tutorials} component={TutorialsPage} key="TutorialsPage" />,
+    <Route exact path={docs.visualEditor} component={VisualEditorPage} key="VisualEditorPage" />,
+    <Route exact path={docs.streamrEngine} component={StreamrEnginePage} key="StreamrEnginePage" />,
+    <Route exact path={docs.dataMarketplace} component={MarketplacePage} key="MarketplacePage" />,
+    <Route exact path={docs.api} component={ApiPage} key="ApiPage" />,
+])
+
+const App = () => (
+    <ConnectedRouter history={history}>
+        <ModalRoot>
+            <LocaleSetter />
+            <AutoScroll />
+            <Switch>
+                <Route exact path={routes.logout()} component={LogoutPage} />
+                <Route exact path={formatPath(internalLogin, ':type?')} component={LoginRedirect} />
+                <Route exact path={links.root} component={Products} />
+                <Route exact path={marketplace.main} component={Products} />
+                <Route exact path={links.marketplace.createProduct} component={CreateProductAuth} />
+                <Route exact path={formatPath(marketplace.products, ':id', 'purchase')} component={ProductPurchasePage} />
+                <Route exact path={formatPath(marketplace.products, ':id', 'publish')} component={ProductPublishPage} />
+                <Route exact path={formatPath(marketplace.products, ':id', 'streamPreview', ':streamId')} component={StreamPreviewPage} />
+                <Route exact path={formatPath(marketplace.products, ':id')} component={ProductPage} />
+                <Route exact path={routes.editProduct()} component={EditProductAuth} />
+                {DocsRouter()}
```

Comment:
Should we layout the other components in a similar way? Ie.

```
<Switch>
    {MarketplaceRouter()}   
    {DocsRouter()}
    {UserPagesRouter()}
</Switch>
```

Reply:
Thanks for reminding me about this - I think so, I will adapt if possible. 

There was an issue with arranging the userpages routes in this sort of array but I think it was related to the StreamrClient Provider wrapping which isn't in use anymore.

Reply:
This would be nice 👍